adapter/service/outlvlswitch/routing.go

```diff
@@ -41,6 +44,90 @@ func RouteCmdLvlSet(serviceRegistry adapter.ServiceRegistry) *router.Routing {
 	)
 }
 
+// RouteCmdLvlStart returns a routing responsible for handling the command.
+func RouteCmdLvlStart(serviceRegistry adapter.ServiceRegistry) *router.Routing {
+	return router.NewRouting(
+		HandleCmdLvlStart(serviceRegistry),
+		router.ForService(OutLvlSwitch),
+		router.ForType(CmdLvlStart),
+	)
+}
+
+// RouteCmdLvlStop returns a routing responsible for handling the command.
+func RouteCmdLvlStop(serviceRegistry adapter.ServiceRegistry) *router.Routing {
+	return router.NewRouting(
+		HandleCmdLvlStop(serviceRegistry),
+		router.ForService(OutLvlSwitch),
+		router.ForType(CmdLvlStop),
+	)
+}
+
+// HandleCmdLvlStart returns a handler responsible for handling CmdLvlStart message.
+func HandleCmdLvlStart(serviceRegistry adapter.ServiceRegistry) router.MessageHandler {
+	return router.NewMessageHandler(
+		router.MessageProcessorFn(func(message *fimpgo.Message) (*fimpgo.FimpMessage, error) {
+			s := serviceRegistry.ServiceByTopic(message.Topic)
+			if s == nil {
+				return nil, fmt.Errorf("adapter: service not found under the provided address: %s", message.Addr.ServiceAddress)
+			}
+
+			service, ok := s.(Service)
+			if !ok {
+				return nil, fmt.Errorf("adapter: incorrect service found under the provided address: %s", message.Addr.ServiceAddress)
+			}
+
+			direction, err := message.Payload.GetStringValue()
+			if err != nil {
+				return nil, errors.Wrap(err, "adapter: error while getting level value from message")
+			}
+
+			duration, err := getDurationInSeconds(message)
+			if err != nil {
+				return nil, errors.Wrap(err, "adapter: error while getting duration value from message")
+			}
+
+			startLvl, err := getStartLvl(message)
+			if err != nil {
+				return nil, errors.Wrap(err, "adapter: error while getting start_lvl value from message")
+			}
+
+			if err := service.StartLevelTransition(direction, startLvl, duration); err != nil {
```

Comment:
I think there is issue here. If user sent command without startLvl but device supports it, the transition will start from 0 level, instead of current level.

Please rewrite this routing. Use your defined object "LevelTransitionParams". 

Set it accordingly whether the parameters are present or not. Return invalid format error if they are present but cannot be parsed as integers.

Pass to the service method direction and LevelTransitionParams.